tests/unit/s2n_client_key_share_extension_pq_test.c

```diff
@@ -349,6 +348,76 @@ int main() {
+
+                        const struct s2n_ecc_preferences *ecc_preferences = NULL;
+                        EXPECT_SUCCESS(s2n_connection_get_ecc_preferences(conn, &ecc_preferences));
+                        EXPECT_NOT_NULL(ecc_preferences);
+
+                        const struct s2n_kem_preferences *kem_pref = NULL;
+                        EXPECT_SUCCESS(s2n_connection_get_kem_preferences(conn, &kem_pref));
+                        EXPECT_NOT_NULL(kem_pref);
+
+                        struct s2n_stuffer first_extension = { 0 }, second_extension = { 0 };
+                        EXPECT_SUCCESS(s2n_stuffer_growable_alloc(&first_extension, MEM_FOR_EXTENSION));
+                        EXPECT_SUCCESS(s2n_stuffer_growable_alloc(&second_extension, MEM_FOR_EXTENSION));
+
+                        EXPECT_SUCCESS(s2n_client_key_share_extension.send(conn, &first_extension));
+
+                        conn->secure.server_kem_group_params.kem_group = conn->secure.client_kem_group_params[0].kem_group;
+                        conn->secure.server_kem_group_params.ecc_params.negotiated_curve =
+                                conn->secure.client_kem_group_params[0].ecc_params.negotiated_curve;
+
+                        /* Setup the client to have received a HelloRetryRequest */
+                        EXPECT_MEMCPY_SUCCESS(conn->secure.server_random, hello_retry_req_random, S2N_TLS_RANDOM_DATA_LEN);
+                        EXPECT_SUCCESS(s2n_connection_set_all_protocol_versions(conn, S2N_TLS13));
+                        EXPECT_SUCCESS(s2n_set_connection_hello_retry_flags(conn));
+                        conn->early_data_state = S2N_EARLY_DATA_REJECTED;
+
+                        EXPECT_SUCCESS(s2n_client_key_share_extension.send(conn, &second_extension));
+
+                        uint16_t first_sent_key_shares_size = 0, second_sent_key_shares_size = 0;
+                        EXPECT_SUCCESS(s2n_stuffer_read_uint16(&first_extension, &first_sent_key_shares_size));
+                        EXPECT_SUCCESS(s2n_stuffer_read_uint16(&second_extension, &second_sent_key_shares_size));
+                        EXPECT_EQUAL(first_sent_key_shares_size, s2n_stuffer_data_available(&first_extension));
+                        EXPECT_EQUAL(second_sent_key_shares_size, s2n_stuffer_data_available(&second_extension));
+                        EXPECT_TRUE(second_sent_key_shares_size < first_sent_key_shares_size);
+
+                        uint16_t first_sent_hybrid_iana_id = 0, second_sent_hybrid_iana_id = 0;
+                        EXPECT_SUCCESS(s2n_stuffer_read_uint16(&first_extension, &first_sent_hybrid_iana_id));
+                        EXPECT_SUCCESS(s2n_stuffer_read_uint16(&second_extension, &second_sent_hybrid_iana_id));
+                        EXPECT_EQUAL(first_sent_hybrid_iana_id, conn->secure.client_kem_group_params[0].kem_group->iana_id);
+                        EXPECT_EQUAL(second_sent_hybrid_iana_id, second_sent_hybrid_iana_id);
```

Comment:
What is line 395 supposed to be testing?

Reply:
Ahaha absolutely nothing. Good catch!